test/error_messages.coffee

```diff
@@ -942,14 +942,14 @@ test "#4130: unassignable in destructured param", ->
   '''
 
 test "`yield` outside of a function", ->
-  assertErrorFormat '''
+  assertErrorFormatAst '''
```

Comment:
Why are we changing these tests into `assertErrorFormatAst`? The same error should still be thrown for regular code being compiled into JS.

Reply:
`assertErrorFormatAst()` tests that *both* compile-to-JS and AST generation throw the error. I can see how the name might imply *just* AST, but a name that is more explicit that it's testing both would probably be really long?

Reply:
Yeah . . . `assertErrorFormatJsAndAst`? Or make the combined check the default, and JS only an option?

Reply:
I like the idea of making the combined the default behavior of `assertErrorFormat()`

I think doing that refactor after we've gone through all the node classes makes sense since then most of them should be able to be tested against both (JS and AST) and it'll probably catch some that should be throwing during AST generation

Reply:
Sounds good.